app/services/pagarme/request/pagarme.rb

```diff
@@ -0,0 +1,51 @@
+class Pagarme::Request::Pagarme
+  PAGARME_URI = ENV.fetch("PAGARME_API_URI")
+  PAGARME_API_KEY = ENV.fetch("PAGARME_API_KEY")
+
+  class << self
+    def post_pagarme_request(request)
+      params =
+        {
+          requests: request,
+          api_key:  PAGARME_API_KEY
+        }.to_json
+
+      headers =
+        {
+          'content-type' => "application/json",
+          'Accept'       => "application/json"
+        }
+
+      HTTParty.post(PAGARME_URI, body: params, headers: headers)
+    end
+
+    def response_formatter(json_objects)
+      formated_json = "[#{json_objects.gsub("\n\u0000", ',')[0..-2]}]"
+
+      JSON.parse(formated_json).map do |subscription|
+        subscription['body'].symbolize_keys unless subscription['body']['errors'].present?
+      end.compact
+    rescue JSON::ParserError
+      Rails.logger.error("Failure to parse the Pagarme response #{json_objects}")
```

Comment:
As discussed before in person, the team should be warned when this occurs or the further subscriptions will never be cancelled, since this is a external format change and requires a code change.